### DISCUSSION TITLE
Optimistic Execution of MSET

### DISCUSSION THREAD

**Discussion by suxb201** [dragonflydb/dragonfly#4018] [Q&A] [answered]
The MSET command comes with the CO::IDEMPOTENT flag, which implies the following scenario:
* k1 = old
* k2 = old

When executing:
* MSET k1 new k2 new
* MGET k1 k2

We expect MGET to return either `old` `old` or `new` `new`. However, is it possible that it could return `old` `new` or `new` `old`?

Consider the following execution sequence (assuming no inline execution):
* shard1: MGET k1(in txq); MSET k1(enqueue)
* shard2: MSET k2(optimistic); MGET k2(enqueue)

In this scenario, MGET might return `old` `new`. Is this correct, or am I misunderstanding something?

**Comment by romange** [comment, marked as answer]
You are absolutely right with you question, the only thing is that `MSET` does not have `CO::IDEMPOTENT` flag. MGET has :)
